app/src/test/java/org/mozilla/focus/browser/integration/FullScreenIntegrationTest.kt

```diff
@@ -95,61 +99,58 @@ internal class FullScreenIntegrationTest {
         val activity: Activity = mock()
         doReturn(activityWindow).`when`(activity).window
         doReturn(decorView).`when`(activityWindow).decorView
-        val integration = FullScreenIntegration(
-            activity, mock(), null, mock(), mock(), mock(), mock(), mock()
+
+        val integration = spy(
+            FullScreenIntegration(
+                activity, mock(), null, mock(), mock(), mock(), mock(), mock()
+            )
         )
 
         integration.switchToImmersiveMode()
 
-        verify(activityWindow).addFlags(FLAG_KEEP_SCREEN_ON)
-        verify(decorView).systemUiVisibility =
-            View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION or
-            View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or
-            View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+        verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
+        verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+        // verify that the immersive mode restoration is set as expected
+        verify(decorView).setOnApplyWindowInsetsListener(any())
     }
 
     @Test
     @Suppress("DEPRECATION")
     fun `GIVEN immersive mode WHEN exitImmersiveModeIfNeeded is called THEN show the system bars`() {
-        val windowAttributes = WindowManager.LayoutParams()
-        // This flag is checked to infer whether already in immersive mode or not
-        windowAttributes.flags = FLAG_KEEP_SCREEN_ON
         val decorView: View = mock()
         val activityWindow: Window = mock()
         val activity: Activity = mock()
         doReturn(activityWindow).`when`(activity).window
         doReturn(decorView).`when`(activityWindow).decorView
-        doReturn(windowAttributes).`when`(activityWindow).attributes
-        val integration = FullScreenIntegration(
-            activity, mock(), null, mock(), mock(), mock(), mock(), mock()
+
+        val integration = spy(
+            FullScreenIntegration(
+                activity, mock(), null, mock(), mock(), mock(), mock(), mock()
+            )
         )
 
         integration.exitImmersiveModeIfNeeded()
 
-        verify(activityWindow).clearFlags(FLAG_KEEP_SCREEN_ON)
-        verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+        verify(decorView, Mockito.times(2)).systemUiVisibility = View.SYSTEM_UI_FLAG_VISIBLE
```

Comment:
```suggestion
        verify(decorView, times(2)).systemUiVisibility = View.SYSTEM_UI_FLAG_VISIBLE
```

Just curious, do you know why the above call is made twice? If so it may worth a comment above the line.

Reply:
setSystemUiVisibility will be called twice by WindowInsetsControllerCompat once for setting the status bar and another for setting the navigation bar. Will add the comment